protocol/invocation/rpcinvocation.go

```diff
@@ -111,6 +113,25 @@ func (r *RPCInvocation) AttachmentsByKey(key string, defaultValue string) string
 	return defaultValue
 }
 
+// get attributes
+func (r *RPCInvocation) Attributes() map[string]interface{} {
+	return r.attributes
+}
+
+// get attribute by key. If it is not exist, it will return default value
+func (r *RPCInvocation) AttributeByKey(key string, defaultValue interface{}) interface{} {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+	if r.attributes == nil {
```

Comment:
after init this map in newRPCInvocation, u can delete this if condition clause.

Reply:
There are two methods about NewRPCInvocation . Another one is NewRPCInvocationWithOptions.  We cannot make it init it well.

Reply:
I think we could init the attributes in NewRPCInvocationWithOptions too. The attribute map is empty. What's your concern?

Reply:
@flycash can not agree with u more.

Reply:
@cvictory pls improve NewRPCInvocationWithOptions.

Reply:
I have added WithAttributes function for it.

Reply:
@cvictory pls delete WithAttributes and modify NewRPCInvocationWithOptions